src/XrdHttp/XrdHttpUtils.hh

```diff
@@ -79,15 +83,97 @@ bool Fromhexdigest(const unsigned char *input, int length, unsigned char *out);
 
 void Tobase64(const unsigned char *input, int length, char *out);
 
-
 // Create a new quoted string
 char *quote(const char *str);
-
 // unquote a string and return a new one
 char *unquote(char *str);
 
-
-
+/**
+ * Creates a non-const copy of the string passed in parameter and
+ * calls unquote() on it before returning the pointer
+ * to the unquoted string
+ * @param str the string to unquote
+ * @return the malloc'd and unquoted string
+ * !!! IT MUST BE FREED AFTER USAGE USING free(...) !!!
+ */
+inline char * decode_raw(const char * str) {
+  if(!str) {
+    return nullptr;
+  }
+  size_t strLength = strlen(str);
+  // uniquely own the temporary copy
+  std::unique_ptr<char[]> buf(new char[strLength + 1]);
+  std::memcpy(buf.get(), str, strLength + 1);
+  // unquote returns a fresh malloc()'d pointer
+  return unquote(buf.get());
+}
+
+/**
+ * Calls quote() on the string passed in parameter
+ * @param str the string to quote
+ * @return the pointer to the quoted string
+ * !!! IT MUST BE FREED AFTER USAGE USING free(...) !!!
+ */
+inline char * encode_raw(const char * str) {
+  if(!str) {
+    return nullptr;
+  }
+  return quote(str);
+}
+
+/**
+ * Encodes the URL passed in parameter (converts all letters consider illegal in URLs to their
+ * %XX versions).
+ * Calls quote()
+ * Returns a std::string containing the encoded string
+ */
+inline std::string encode_str(const char * str) {
```

Comment:
I'd take a reference to a string here as input, to avoid having to deal with `nullptr`. Same for the other functions.

Reply:
I'm modifying it but this means that the user should check if the char * passed is nullptr or not (one cannot construct a string with a nullptr).

Reply:
It's already the case in the code so I modified it!